client/src/components/DataFiles/DataFilesModals/DataFilesExtractModal.js

```diff
@@ -0,0 +1,104 @@
+import React, { useMemo } from 'react';
+import { useSelector, useDispatch, shallowEqual } from 'react-redux';
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
+import { LoadingSpinner, Icon, InlineMessage } from '_common';
+import { useHistory, useLocation } from 'react-router-dom';
+import './DataFilesCompressModal.module.scss';
+
+const DataFilesExtractModal = () => {
+  const history = useHistory();
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const status = useSelector(
+    state => state.files.operationStatus.extract,
+    shallowEqual
+  );
+
+  const params = useSelector(
+    state => state.files.params.FilesListing,
+    shallowEqual
+  );
+
+  const isOpen = useSelector(state => state.files.modals.extract);
+  const selectedFiles = useSelector(({ files: { selected, listing } }) =>
+    selected.FilesListing.map(i => ({
+      ...listing.FilesListing[i]
+    }))
+  );
+  const selected = useMemo(() => selectedFiles, [isOpen]);
+
+  const toggle = () =>
+    dispatch({
+      type: 'DATA_FILES_TOGGLE_MODAL',
+      payload: { operation: 'extract', props: {} }
+    });
+
+  const onOpened = () => {
+    dispatch({
+      type: 'FETCH_FILES_MODAL',
+      payload: { ...params, section: 'modal' }
+    });
+  };
+
+  const onClosed = () => {
+    dispatch({ type: 'DATA_FILES_MODAL_CLOSE' });
+    if (status) {
+      dispatch({
+        type: 'DATA_FILES_SET_OPERATION_STATUS',
+        payload: { status: {}, operation: 'extract' }
+      });
+      dispatch({ type: 'GET_JOBS', params: { offset: 0 } });
```

Comment:
Do we need these dispatches? I think the jobs list will refresh dynamically as we get webhooks back from tapis. Or is that taking too long?

Reply:
My mistake, I noticed that the dashboard job listing wasn't updating. This was because ngrok wasn't running